components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -125,7 +125,7 @@ func (n *Nats) SyncSubscription(sub *eventingv1alpha1.Subscription, cleaner even
 // DeleteSubscription deletes all NATS subscriptions corresponding to a Kyma subscription
 func (n *Nats) DeleteSubscription(subscription *eventingv1alpha1.Subscription) error {
 	for key, sub := range n.subscriptions {
-		if strings.HasPrefix(key, createKeyPrefix(subscription)) {
+		if createKeyPrefix(subscription) == createKymaSubscriptionNamespacedName(key, sub).String() {
```

Comment:
My proposal: put function into external function and add some table driven tests. I am willing to contribute them